hs_core/views/__init__.py

```diff
@@ -593,14 +593,32 @@ def add_metadata_element(request, shortkey, element_name, *args, **kwargs):
             res.update_public_and_discoverable()
             resource_modified(res, request.user, overwrite_bag=False)
     else:
+
+        if element_name.lower() == "subject":
+            original_value = request.POST.get("value", "")
+            keywords = original_value.replace(",", "")
+            # Make request.POST mutable
+            mutable_post = request.POST.copy()
+            mutable_post["value"] = keywords
+            request.POST = mutable_post
+
         handler_response = signals.pre_metadata_element_create.send(
             sender=sender_resource, element_name=element_name, request=request
         )
+
+        if element_name.lower() == "subject" and original_value:
+            mutable_post["value"] = original_value  # Reassign original value
+            request.POST = mutable_post
+
         for receiver, response in handler_response:
             if "is_valid" in response:
                 if response["is_valid"]:
                     element_data_dict = response["element_data_dict"]
+
                     if element_name == "subject":
+                        if original_value:
+                            element_data_dict["value"] = original_value
+
```

Comment:
It looks here like you're trying to handle comma delimited entries.
I'm not convinced that you need to modify this `__init__.py` file at all.
I think that if you modify your `newKeywordsLength` function so that it is accurate, you can remove all of the modifications that you are making here.

Reply:
Perhaps I'm missing what you're trying to do here. It looks to me like your trying to handle comma separated values. Is that the case? If yes, shouldn't it already be handled: https://github.com/hydroshare/hydroshare/blob/2.15.10/hs_core/views/__init__.py#L606
(when value is POSTed as a comma-separated string)?